beekeeper-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/generator/ExpiredHousekeepingMetadataGenerator.java

```diff
@@ -74,17 +91,46 @@ public LifecycleEventType getLifecycleEventType() {
     return LIFECYCLE_EVENT_TYPE;
   }
 
-  private HousekeepingEntity generateHousekeepingEntity(ListenerEvent listenerEvent, String clientId) {
+  private HousekeepingEntity generateHousekeepingEntity(CreateTableEvent listenerEvent, String clientId) {
+    return generateHousekeepingEntity(listenerEvent, clientId, listenerEvent.getTableLocation(), null);
+  }
+
+  private HousekeepingEntity generateHousekeepingEntity(AlterTableEvent listenerEvent, String clientId) {
+    return generateHousekeepingEntity(listenerEvent, clientId, listenerEvent.getTableLocation(), null);
+  }
+
+  private HousekeepingEntity generateHousekeepingEntity(AddPartitionEvent listenerEvent, String clientId) {
+    String partitionName = generatePartitionName(Lists.newArrayList(listenerEvent.getPartitionKeys().keySet()),
+        listenerEvent.getPartitionValues());
+    return generateHousekeepingEntity(listenerEvent, clientId, listenerEvent.getPartitionLocation(), partitionName);
+  }
+
+  private HousekeepingEntity generateHousekeepingEntity(AlterPartitionEvent listenerEvent, String clientId) {
+    String partitionName = generatePartitionName(Lists.newArrayList(listenerEvent.getPartitionKeys().keySet()),
+        listenerEvent.getPartitionValues());
+    return generateHousekeepingEntity(listenerEvent, clientId, listenerEvent.getPartitionLocation(), partitionName);
```

Comment:
This is because database and table name are common to all events but the path is unique to the type of ListenerEvent. For example, from `AddPartitionEvent` we need partition location but from `AlterTableEvent` we need table location.